client/checkout/blocks/upe-fields.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * External dependencies
+ */
+import {
+	Elements,
+	ElementsConsumer,
+	PaymentElement,
+} from '@stripe/react-stripe-js';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import confirmUPEPayment from './confirm-upe-payment.js';
+import { getConfig } from 'utils/checkout';
+import { PAYMENT_METHOD_NAME_CARD } from '../constants.js';
+
+const WCPayUPEFields = ( {
+	api,
+	activePaymentMethod,
+	stripe,
+	elements,
+	billing: { billingData },
+	eventRegistration: {
+		onPaymentProcessing,
+		onCheckoutAfterProcessingWithSuccess,
+	},
+	emitResponse,
+} ) => {
+	const [ paymentIntentId, setPaymentIntentId ] = useState( null );
+	const [ clientSecret, setClientSecret ] = useState( null );
+	const [ hasRequestedIntent, setHasRequestedIntent ] = useState( false );
+	const [ isUPEComplete, setIsUPEComplete ] = useState( false );
+
+	const testMode = getConfig( 'testMode' );
+	const testCopy = (
+		<p>
+			<strong>Test mode:</strong> use the test VISA card 4242424242424242
+			with any expiry date and CVC.
+		</p>
+	);
+
+	useEffect( () => {
+		if ( paymentIntentId || hasRequestedIntent ) {
```

Comment:
Is there a better way to create the intention on component load or maybe even before that?

Reply:
I'm not sure if there's a better React-y way to do this, but this seems to me like the appropriate place to keep this hook for this purpose.